ivy/functional/frontends/jax/random.py

```diff
@@ -283,3 +283,21 @@ def pareto(key, b, shape=None, dtype="float64"):
     e = -ivy.log(1 - uniform)
 
     return ivy.exp(e / b)
+
+
+@handle_jax_dtype
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes(
+    {
+        "0.4.11 and below": (
+            "float16",
+            "bfloat16",
+        )
+    },
+    "jax",
+)
+def ball(key, d, p=2, shape=(), dtype="float64"):
+    seed = _get_seed(key)
+    g = generalized_normal(seed[0], p, (*shape, d), dtype)
+    e = exponential(seed[1], shape, dtype)
+    return g / (((ivy.np.abs(g) ** p).sum(-1) + e) ** (1 / p))[..., None]
```

Comment:
We don't call `ivy.np` all functions from our functional API should be directly accessed from `ivy`